`tests/arch/arm/arm_irq_vector_table` fails to build when intc driver directly uses `_sw_isr_table`

**Describe the bug**

Tested on 4d0a8c1e8b8316ec5071a3fcc98cd292d3bc4e6d.

`tests/arch/arm/arm_irq_vector_table` fails to build for targets where the interrupt controller driver directly references `_sw_isr_table`. This test n-selects the `CONFIG_GEN_ISR_TABLES` option, causing a build error.

Affected platforms (discovered so far):
* `arduino_uno_r4_wifi`
* `mikroe_clicker_ra4m1`

The interrupt controller driver in question: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/interrupt_controller/intc_renesas_ra_icu.c.

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b arduino_uno_r4_wifi tests/arch/arm/arm_irq_vector_table`
2. See error

**Expected behavior**
The test can be successfully built for said platforms.

**Impact**
Showstopper (discovered in #74390).

**Logs and console output**
```
/home/filip/zephyrproject/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a(intc_renesas_ra_icu.c.obj): in function `ra_icu_query_available_irq':
/home/filip/zephyrproject/zephyrproject/zephyr/drivers/interrupt_controller/intc_renesas_ra_icu.c:47: undefined reference to `_sw_isr_table'
/home/filip/zephyrproject/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a(intc_renesas_ra_icu.c.obj): in function `ra_icu_query_irq_config':
/home/filip/zephyrproject/zephyrproject/zephyr/drivers/interrupt_controller/intc_renesas_ra_icu.c:73: undefined reference to `_sw_isr_table'
/home/filip/zephyrproject/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a(intc_renesas_ra_icu.c.obj): in function `ra_icu_irq_connect_dynamic':
/home/filip/zephyrproject/zephyrproject/zephyr/drivers/interrupt_controller/intc_renesas_ra_icu.c:99: undefined reference to `z_isr_install'
/home/filip/zephyrproject/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a(intc_renesas_ra_icu.c.obj): in function `ra_icu_irq_disconnect_dynamic':
/home/filip/zephyrproject/zephyrproject/zephyr/drivers/interrupt_controller/intc_renesas_ra_icu.c:118: undefined reference to `z_isr_install'
Memory region         Used Size  Region Size  %age Used
           FLASH:       26913 B       240 KB     10.95%
             RAM:        5536 B        32 KB     16.89%
        IDT_LIST:          0 GB        32 KB      0.00%
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: 4d0a8c1e8b8316ec5071a3fcc98cd292d3bc4e6d.

--- a/drivers/interrupt_controller/Kconfig.renesas_ra
+++ b/drivers/interrupt_controller/Kconfig.renesas_ra
@@ -5,5 +5,6 @@ config RENESAS_RA_ICU
 	bool "Renesas RA series interrupt controller unit"
 	default y
 	depends on DT_HAS_RENESAS_RA_INTERRUPT_CONTROLLER_UNIT_ENABLED
+	select GEN_ISR_TABLES
 	help
 	  Renesas RA series interrupt controller unit

--- a/tests/arch/arm/arm_irq_vector_table/testcase.yaml
+++ b/tests/arch/arm/arm_irq_vector_table/testcase.yaml
@@ -7,7 +7,7 @@ common:
   arch_allow: arm
 tests:
   arch.arm.irq_vector_table:
-    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
+    filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE and
+            not CONFIG_GEN_ISR_TABLES
     platform_exclude:
       - mr_canhubk3
-      - arduino_uno_r4_minima

